File: ruby.vb.ns.com/scripts/example/method.rb
#
# メソッド
#

### ! 破壊的メソッド
# 元の値を上書きする
puts "---- 破壊的メソッド ----"

s = "hello"

puts s.upcase  # HELLO
puts s         # hello

puts s.upcase! # HELLO
puts s         # HELLO

### ? 真偽値を返すメソッド
puts "---- 真偽値メソッド ----"

s = ""
p s.empty? # true

s = "hello"
p s.empty? # false

### 関数的メソッド
puts "---- 関数的メソッド ----"

def myfunc(name = "hoge")
    s = "hello " + name
    return s
end

puts myfunc("foo")
puts myfunc("bar")
puts myfunc()

File: java.vb.ns.com/scripts/example/Method.java
//
// 関数
//
public class Method {
    public static void main(String[] args) {
        System.out.println("---- method ----");

        foo("bar");
        foo();
    }

    public static void foo(String val) {
        System.out.println("foo" + val);
    }

    // オーバーロード
    public static void foo() {
        System.out.println("foo overload");
    }
}
